Automattic/jetpack
# Social: Add social note toggle admin page

This change adds a toggle to the Social admin page, so users can toggle on/off the Social Notes feature and custom post type.

It should be easier to review by commits as described below.

## Proposed changes:

* Added the jetpack endpoints, and modified the setter function 24e614dbaf165e801d21c3904c6d21778e0713bb
* Added the initial state 95cc4b596ff367226c5cf73a26e5a1e70ad8ce13
* Prepared the React store to be able to fetch and update the new endpoints 4264b4bb8c70c0293955935a182f7d63e55da213
* Added a new beta property to the ToggleComponent b8b1517e1d4343f84e428ee280e0273fec10890d
* Created the actual React component for the toggle 02f54826db0149fd26a1d1bfe2ef9d780a020a35

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/208
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Check out this PR and go to the Social admin page.
* You should see the new toggle for the Social Notes.
  * It should be off for new sites, and you should not see the Social Notes menu
  * Toggling it should put the slider into an infinite disabled state until the page reloads.
  * After reloading it should reflect the correct state, and the menu item should appear/disappear

![CleanShot 2024-02-14 at 14 59 41 png](https://github.com/Automattic/jetpack/assets/36671565/03ba8c7b-f61c-498d-b3b1-06cf2f42ec46)
